x/programs/runtime/import_program.go

```diff
@@ -63,7 +64,8 @@ func NewProgramModule(r *WasmRuntime) *ImportModule {
 				return result, nil
 			})},
 			"set_call_result": {FuelCost: setResultCost, Function: FunctionNoOutput(func(callInfo *CallInfo, input []byte) error {
-				callInfo.inst.result = input
+				// needs to clone because this points into the current store's linear memory which may be gone when this is read
+				callInfo.inst.result = slices.Clone(input)
```

Comment:
Isn't `slices.Clone` doing a shallow copy ? So maybe that the pointer is not cloned

Reply:
Elements are bytes in this case, so shallow copy of the bytes is fine.